src/main/java/com/example/Restaurant_Management_System_REST_API/service/RestaurantCustomerDetailsService.java

```diff
@@ -0,0 +1,27 @@
+package com.example.Restaurant_Management_System_REST_API.service;
+
+import com.example.Restaurant_Management_System_REST_API.model.entity.Customer;
+import com.example.Restaurant_Management_System_REST_API.repository.CustomerRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class RestaurantCustomerDetailsService implements UserDetailsService {
+
+    private final CustomerRepository customerRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+
+        Customer customer = customerRepository.findByEmailAddress(email).orElseThrow(() ->
+                new UsernameNotFoundException("Username with "
+                        + email + " was not found!"));
+
+        return new org.springframework.security.core.userdetails.User(customer.getUsername(), customer.getPassword(), customer.getAuthorities());
```

Comment:
Could you describe for me please why it's returns like that and what's the concept of that line with return? I mean what's the reason behind making it that way. Is there another possibility of writing this code?